torch/lib/c10d/logger.hpp

```diff
@@ -21,14 +21,37 @@ class Logger {
   void set_env_variables();
   // Set parameters stats.
   void set_parameter_stats();
-  // Set buckets related stats.
-  void set_bucket_stats();
+  // Get size of each bucket (Bytes).
+  std::vector<int> get_bucket_sizes();
+
+  // Calculate avg stats using cpu timer and gpu timer
+  // that has been recorded in reducer.
+  void calculate_avg_cpu_time(
+      int64_t& avg_time,
+      int64_t cpu_start_time,
+      int64_t cpu_end_time);
+#ifdef USE_CUDA
+  void calculate_avg_gpu_time(
+      int64_t& avg_time,
+      at::cuda::CUDAEvent& gpu_start,
+      at::cuda::CUDAEvent& gpu_end);
+#endif
+  // Set stats that can be collected only during
+  // training loop. It is called at the beginning of forward call
+  // to record the run time stats of iterations previouly ran.
+  // GPU performance stats are collected only for single process
+  // single device program right now.
+  // TODO to support single process multiple devices, events need
+  // to be created and recorded on multiple devices.
+  void set_runtime_stats();
```

Comment:
Nit: document that this is now sampled? 

Also, if `set_runtime_stats` is now sampled, then what happens if I call `model_ddp.logger.get_logging_data()` and view the runtime stats, on, say iteration 30? Will the runtime stats correspond to the most recently sampled iteration (i.e.  first iteration in this case)? Is there any way to indicate to the user that they are not necessarily getting most recent data but rather sampled?

Reply:
right, runtime stats will correspond to the most recently sampled iteration. let me clarify it in the comment...